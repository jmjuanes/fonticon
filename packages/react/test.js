import TestRenderer from "react-test-renderer";

import * as Mochicons from "./index.js";

describe("@mochicons/react", () => {
    it("should export all icons as a react components", () => {
        expect(Object.keys(Mochicons)).toMatchSnapshot();
    });

    it("should wrap icons in a <svg> component", () => {
        const iconInstance = TestRenderer.create(<Mochicons.Icon />);

        expect(iconInstance.root.findByType("svg").props.width).toEqual(expect.stringContaining("1em"));
        expect(iconInstance.root.findByType("svg").props.height).toEqual(expect.stringContaining("1em"));
    });

    it("should wrap icons in a <g> component", () => {
        const iconInstance = TestRenderer.create(<Mochicons.Icon />);

        expect(iconInstance.root.findByType("g").props.stroke).toEqual(expect.stringContaining("currentColor"));
        expect(iconInstance.root.findByType("g").props.strokeWidth).toEqual(expect.stringContaining("2"));
    });
});
