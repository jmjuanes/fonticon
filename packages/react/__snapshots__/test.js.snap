// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@mochicons/react should export all icons as a react components 1`] = `
[
  "AddressBookIcon",
  "AlignBottomIcon",
  "AlignCenterIcon",
  "AlignLeftIcon",
  "AlignMiddleIcon",
  "AlignRightIcon",
  "AlignTopIcon",
  "ArchiveIcon",
  "ArrowAltDownIcon",
  "ArrowAltLeftIcon",
  "ArrowAltRightIcon",
  "ArrowAltUpIcon",
  "ArrowCircleDownIcon",
  "ArrowCircleLeftIcon",
  "ArrowCircleRightIcon",
  "ArrowCircleUpIcon",
  "ArrowDownIcon",
  "ArrowDownLeftIcon",
  "ArrowDownRightIcon",
  "ArrowLeftIcon",
  "ArrowRightIcon",
  "ArrowSquareDownIcon",
  "ArrowSquareLeftIcon",
  "ArrowSquareRightIcon",
  "ArrowSquareUpIcon",
  "ArrowUpIcon",
  "ArrowUpLeftIcon",
  "ArrowUpRightIcon",
  "ArrowsIcon",
  "ArrowsMaximizeIcon",
  "ArrowsMinimizeIcon",
  "ArrowsShuffleIcon",
  "ArticleIcon",
  "AspectRatioIcon",
  "AtomIcon",
  "AttachmentIcon",
  "BackIcon",
  "BackpackIcon",
  "BackspaceIcon",
  "BagIcon",
  "BanIcon",
  "BarsIcon",
  "BatteryEmptyIcon",
  "BatteryFullIcon",
  "BatteryHalfIcon",
  "BatteryIcon",
  "BellIcon",
  "BezierIcon",
  "BluetoothIcon",
  "BoldIcon",
  "BoltIcon",
  "BookIcon",
  "BookOpenIcon",
  "BookmarkIcon",
  "BoxIcon",
  "BriefcaseIcon",
  "BrushIcon",
  "BugIcon",
  "CalendarIcon",
  "CameraIcon",
  "CaretDownIcon",
  "CaretLeftIcon",
  "CaretRightIcon",
  "CaretUpIcon",
  "CartIcon",
  "ChartAreaIcon",
  "ChartBarIcon",
  "ChartLineIcon",
  "ChartPieIcon",
  "ChatDotsIcon",
  "ChatIcon",
  "CheckCircleIcon",
  "CheckIcon",
  "CheckSquareIcon",
  "ChevronCircleDownIcon",
  "ChevronCircleLeftIcon",
  "ChevronCircleRightIcon",
  "ChevronCircleUpIcon",
  "ChevronDownIcon",
  "ChevronLeftIcon",
  "ChevronRightIcon",
  "ChevronSquareDownIcon",
  "ChevronSquareLeftIcon",
  "ChevronSquareRightIcon",
  "ChevronSquareUpIcon",
  "ChevronUpIcon",
  "CircleFillIcon",
  "CircleHalfIcon",
  "CircleIcon",
  "ClipboardIcon",
  "ClipboardMinusIcon",
  "ClipboardPlusIcon",
  "ClipboardXIcon",
  "ClockIcon",
  "CloseIcon",
  "CloudIcon",
  "CloudRainIcon",
  "CodeBlockIcon",
  "CodeIcon",
  "CodeSquareIcon",
  "CoffeeIcon",
  "CompassIcon",
  "CookieIcon",
  "CopyIcon",
  "CreditCardIcon",
  "CropIcon",
  "CutIcon",
  "DiamondIcon",
  "DiskIcon",
  "DnaIcon",
  "DollarCircleIcon",
  "DownloadIcon",
  "DrawingIcon",
  "DropletIcon",
  "EditIcon",
  "EmojiConfusedIcon",
  "EmojiEmptyIcon",
  "EmojiExpressionlessIcon",
  "EmojiLaughIcon",
  "EmojiNeutralIcon",
  "EmojiSadIcon",
  "EmojiSmileIcon",
  "EmojiSurpriseIcon",
  "EmojiWinkIcon",
  "EnvelopeIcon",
  "EraseIcon",
  "ExclamationCircleIcon",
  "ExclamationSquareIcon",
  "ExclamationTriangleIcon",
  "ExternalLinkIcon",
  "EyeIcon",
  "EyeSlashIcon",
  "FileCheckIcon",
  "FileDotsIcon",
  "FileEmptyIcon",
  "FileIcon",
  "FileMinusIcon",
  "FilePlusIcon",
  "FileXIcon",
  "FilesIcon",
  "FillIcon",
  "FilterIcon",
  "FlagIcon",
  "FlaskIcon",
  "FolderCheckIcon",
  "FolderIcon",
  "FolderMinusIcon",
  "FolderPlusIcon",
  "FolderXIcon",
  "FoldersIcon",
  "FrameIcon",
  "FrontIcon",
  "FullscreenExitIcon",
  "FullscreenIcon",
  "GiftIcon",
  "GitBranchIcon",
  "GitCommitIcon",
  "GitMergeIcon",
  "GitPullRequestIcon",
  "GlobeIcon",
  "GridIcon",
  "HamburgerIcon",
  "HandGrabIcon",
  "HandPointerIcon",
  "HeadphonesIcon",
  "HeartIcon",
  "HistoryIcon",
  "HistoryRedoIcon",
  "HistoryUndoIcon",
  "HomeIcon",
  "Icon",
  "ImageCircleIcon",
  "ImageIcon",
  "InboxIcon",
  "InfoCircleIcon",
  "InfoSquareIcon",
  "ItalicIcon",
  "KeyboardIcon",
  "LaptopIcon",
  "LayoutIcon",
  "LifebuoyIcon",
  "LightbulbIcon",
  "LineHrIcon",
  "LineIcon",
  "LineVrIcon",
  "LinkIcon",
  "ListIcon",
  "ListSquareIcon",
  "LocationIcon",
  "LockIcon",
  "LoginIcon",
  "LogoutIcon",
  "MapIcon",
  "MegaphoneIcon",
  "MessageIcon",
  "MicrophoneIcon",
  "MinusCircleIcon",
  "MinusIcon",
  "MinusSquareIcon",
  "MobileIcon",
  "MochiIcon",
  "MoneyIcon",
  "MonitorIcon",
  "MoonIcon",
  "MouseIcon",
  "MugIcon",
  "MusicNoteIcon",
  "MusicPlayerIcon",
  "NewsIcon",
  "NoteIcon",
  "NotebookIcon",
  "NotepadIcon",
  "ObjectGroupIcon",
  "ObjectUngroupIcon",
  "PaletteIcon",
  "PathIcon",
  "PauseIcon",
  "PenIcon",
  "PenNibIcon",
  "PencilIcon",
  "PinIcon",
  "PlanetIcon",
  "PlayIcon",
  "PlusCircleIcon",
  "PlusIcon",
  "PlusSquareIcon",
  "PointerIcon",
  "PostcardIcon",
  "PresentationIcon",
  "PrinterIcon",
  "PuzzleIcon",
  "QuestionCircleIcon",
  "QuestionSquareIcon",
  "QuoteIcon",
  "ReceiptIcon",
  "RocketIcon",
  "Ruler2Icon",
  "RulerIcon",
  "ScrollEmptyIcon",
  "ScrollIcon",
  "SearchIcon",
  "SendIcon",
  "ServerIcon",
  "ShapesIcon",
  "ShareIcon",
  "ShieldCheckIcon",
  "ShieldIcon",
  "ShieldXIcon",
  "SlidersIcon",
  "SodaCupIcon",
  "SquareFillIcon",
  "SquareHalfIcon",
  "SquareIcon",
  "StackIcon",
  "StarHalfIcon",
  "StarIcon",
  "StepBackwardIcon",
  "StepForwardIcon",
  "StopIcon",
  "SuitcaseIcon",
  "SunIcon",
  "TableIcon",
  "TabletIcon",
  "TerminalIcon",
  "TextCenterIcon",
  "TextIcon",
  "TextJustifyIcon",
  "TextLeftIcon",
  "TextRightIcon",
  "TicketIcon",
  "TogoCupIcon",
  "ToolIcon",
  "ToolsIcon",
  "TowerIcon",
  "TrashIcon",
  "TriangleIcon",
  "TrophyIcon",
  "TvIcon",
  "TvRetroIcon",
  "UmbrellaIcon",
  "UnderlineIcon",
  "UnlockIcon",
  "UploadIcon",
  "UserCircleIcon",
  "UserIcon",
  "UserMinusIcon",
  "UserPlusIcon",
  "UserXIcon",
  "VideoIcon",
  "VirusIcon",
  "VolumeDownIcon",
  "VolumeMuteIcon",
  "VolumeUpIcon",
  "WalletIcon",
  "WatchIcon",
  "WindowIcon",
  "XCircleIcon",
  "XIcon",
  "XSquareIcon",
  "ZoomInIcon",
  "ZoomOutIcon",
]
`;
